src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsTaggerProvider.cs

```diff
@@ -3,124 +3,116 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.ComponentModel.Composition;
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.EditAndContinue;
-using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.Editor.Tagging;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.CodeAnalysis.Workspaces;
-using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Adornments;
 using Microsoft.VisualStudio.Text.Classification;
-using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Tagging;
 using Microsoft.VisualStudio.Utilities;
-using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.InlineDiagnostics
+namespace Microsoft.CodeAnalysis.Editor.InlineDiagnostics;
+
+[Export(typeof(ITaggerProvider))]
+[ContentType(ContentTypeNames.RoslynContentType)]
+[TagType(typeof(InlineDiagnosticsTag))]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class InlineDiagnosticsTaggerProvider(
+    IThreadingContext threadingContext,
+    IDiagnosticService diagnosticService,
+    IDiagnosticAnalyzerService analyzerService,
+    IGlobalOptionService globalOptions,
+    [Import(AllowDefault = true)] ITextBufferVisibilityTracker? visibilityTracker,
+    IAsynchronousOperationListenerProvider listenerProvider,
+    IEditorFormatMapService editorFormatMapService,
+    IClassificationFormatMapService classificationFormatMapService,
+    IClassificationTypeRegistryService classificationTypeRegistryService)
+    : AbstractDiagnosticsTaggerProvider<InlineDiagnosticsTag>(
```

Comment:
moved to primary constructor.  and now derives from AbstractDiagnosticsTaggerProvider.  AbstractDiagnosticsAdornmentTaggerProvider was merged into AbstractDiagnosticsTaggerProvider